### PR TITLE
Cleanup:  removed broken `Mage_Core_Controller_Front_Router`

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

> Fatal error:  Uncaught Exception: Notice: Array to string conversion  in /var/www/html/app/code/core/Mage/Core/Model/Store.php on line 580

### Manual testing scenarios (*)

```php
$config = new Mage_Core_Model_Config_Element('<?xml version="1.0" encoding="utf-8" ?><test-url />');
$router = new Mage_Core_Controller_Front_Router($config);
var_dump($router->getUrl());
```

### CODE REVIEW

File: app/Mage.php [hunk 1]
```diff
@@ -444,7 +444,7 @@ public static function getStoreConfigFlag($path, $store = null)
     /**
      * Get base URL path by type
      *
-     * @param string $type
+     * @param Mage_Core_Model_Store::URL_TYPE_* $type
```
Comment (kiatng) [inline app/Mage.php:447]: I have not seen this before, not sure if that's a valid doc block. Alternative:
```suggestion
     * @param string $type One of Mage_Core_Model_Store::URL_TYPE_* constants
```

**Comment by kiatng** [comment]
Any idea where is this class `Mage_Core_Controller_Front_Router` reference in the core? I didn't find any, should it be marked `@deprecated`?

File: app/Mage.php (hunk 1 above)
Comment (justinbeaty) [reply app/Mage.php:447]: It seems that it is supported https://phpstan.org/writing-php-code/phpdoc-types#literals-and-constants

Comment (kiatng) [reply app/Mage.php:447]: Thanks, learn something new.

**Review by kiatng — DISMISSED** [review]

**Comment by sreichel** [comment]
> should it be marked `@deprecated`?

Better to remove it? It was a fatal error in any case ... not usable at all.

**Comment by kiatng** [comment]
Yes, it's not functional, make sense to remove.

**Comment by kiatng** [comment]
I was thinking removing the entire file.

**Review by kiatng — APPROVED** [review]

**Review by addison74 — APPROVED** [review]

**Review by kiatng — APPROVED** [review]

**Pull request merged**